primitives/src/asset.rs

```diff
@@ -0,0 +1,61 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use codec::{Decode, Encode};
+use polymesh_primitives_derive::VecU8StrongTyped;
+use sp_std::prelude::Vec;
+
+/// A wrapper for a token name.
+#[derive(
+    Decode, Encode, Clone, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord, VecU8StrongTyped,
+)]
+pub struct AssetName(pub Vec<u8>);
+
+/// The type of an asset represented by a token.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub enum AssetType {
+    /// Common stock.
+    EquityCommon,
+    /// Preferred stock.
+    EquityPreferred,
+    /// Commodity.
+    Commodity,
+    /// Fixed income security, for example, a bond.
```

Comment:
If possible, could you try to elaborate the docs on the other variants too, giving examples etc.? For someone who understands finance all of this will be clear, but not so for programmers in general I think, so elaborating on the domain is useful I think.

Reply:
Done.